os/compression/compress_read.c

```diff
@@ -73,8 +79,22 @@ static int compress_decompress_block(unsigned char *out_buffer, unsigned int *wr
 			ret = -ret;
 		}
 	}
+#elif CONFIG_COMPRESSION_TYPE == 2
+		if (compression_header->compression_format == COMPRESSION_TYPE_MINIZ) {
+			/* Miniz specific logic for decompression */
+			*writesize = compression_header->blocksize;
+	
+			ret = mz_uncompress(out_buffer, writesize, read_buffer, *size);
+	
+			if (ret != Z_OK) {
+				bmdbg("Failure to decompress with Miniz's uncompress API; ret = %d\n", ret);
```

Comment:
`bmdbg` is for binary manager debug. Let's declare separate debug option.

Reply:
Done. Added separate debug option for Binary Compression. Changes added in a separate commit.